DOCS-6157-added-proxy-install (#13)
mulesoft/docs-service-mesh: modules/ROOT/pages/download-and-install-service-mesh.adoc

== Install Anypoint Service Mesh

> When you invoke the installer, Istio, Service Catalog, Service Mesh Broker, and ClusterIP Service components are installed in your Kubernetes cluster. The installer performs several steps as part of the installation, such as checks for dependencies and performs various configurations. If you failed to complete any of the prerequisites listed in the xref:prepare-to-install-service-mesh.adoc[prepare to install] section, your installation might fail.

Reviewer: ```suggestion
When you invoke the installer, Istio, Service Catalog, Service Mesh Broker, and ClusterIP Service components are installed in your Kubernetes cluster. The installer performs several steps as part of the installation, such as checking for dependencies and performing various configurations. If you failed to complete any of the prerequisites listed in the xref:prepare-to-install-service-mesh.adoc[prepare to install] section, your installation might fail.
```
MIght it fail, or DOES it fail?

Author: Done! Thanks!

> === Install With No Proxy

Reviewer: ```suggestion
=== Install Without Proxy
```

Author: Done!

To install Anypoint Service Mesh, perform the following steps:

> . From the location where you downloaded your Service Mesh client, type:

Reviewer: ```suggestion
. In the location from which you downloaded your Anypoint Service Mesh client, type:
```
GLOBALLY search for and replace "Service Mesh" with the complete product name. There is no approved abbreviated version of it.

Reviewer: HIDDEN TEXT edit
Please compare *** with the original; I cannot edit right in the file of any hidden text.

./service-mesh install
+
The Welcome to Service Mesh installer message is displayed. ADDED BACKTICKS
. Enter the client ID that you obtained as part of performing the prerequisites. EDITED
. Enter the client secret that you obtained as part of performing the prerequisites. EDITED
. Enter the MuleSoft that you obtained by emailing the MuleSoft team as part of performing the prerequisites.  ***IS THIS A LICENSE NUMBER OR A PATH, AS SPECIFIED LATER?***
+
Anypoint Service Mesh installs within few minutes and displays a status:
+
image::service-mesh-installer-output.png[Example Installation Output]
+
If an installation error occurs, the installer times out and uninstalls all previously installed Service Mesh components.
. Verify the status of all pods:
+
kubectl get pods -n service-mesh
+
Anypoint Service Mesh is installed and the status of all pods is displayed: EDITED

+
image::verified-service-mesh-installer-output.png[]

> === Install With Proxy

Reviewer: ```suggestion
=== Install with Proxy
```

Author: Done! Thanks!

> You might need to install Anypoint Service Mesh in a proxy environment. In such cases, you must provide provide the proxy configuration during the installation.

Reviewer: ```suggestion
If you need to install Anypoint Service Mesh in a proxy environment, provide the proxy configuration during the installation.
```

Author: Done! Thanks!

> To install Anypoint Service Mesh with proxy configuration:

Reviewer: OMIT
```suggestion
```

Author: Done! Thanks!

> . From the location where you downloaded your Service Mesh client, type:

Reviewer: ```suggestion
. In the location from which you downloaded your Anypoint Service Mesh client, type:
```

Author: Done! Thanks!

+
`./service-mesh install --clientId=<clientId> --clientSecret=<clientSecret> --license=<license absolute or relative path> --platformUri=<Anypoint Platform Uri> --proxy.host=<host> --proxy.protocol=<protocol> --proxy.port=<port> --proxy.user=<proxy username> --proxy.password=<proxy password>`
+
[%header%autowidth.spread]
|===
| Parameter Name | Description | Required/Optional | Default Value

> | `clientId` | The client ID of the environment (recommended) or the organization. Use the client ID that you obtained as part of the permission and roles prerequisites as explained in this document. | Required | N/A

Reviewer: ```suggestion
| `clientId` | The client ID of the environment (recommended) or the organization. Use the client ID that you obtained as part of the permission and roles prerequisites. | Required | None
```
GLOBALLY replace "N/A" with "none" or "Not applicable." If you must abbr., use "n.a."

Author: Done! Thanks!

> | `clientSecret` | The client secret of the environment (recommended) or the organization. Use the client secret that you obtained as part of the permission and roles prerequisites as explained in this document.| Required | N/A

Reviewer: ```suggestion
| `clientSecret` | The client secret of the environment (recommended) or the organization. Use the client secret that you obtained as part of performing the permission and roles prerequisites.| Required | None
```

Author: Done!

> | `license` | The Mule license you obtained as part of the prerequisites. Provide the path to the location where you license is stored. | Required | N/A

Reviewer: ```suggestion
| `license` | The path to the location of the MuleSoft license you obtained as part of performing the prerequisites. | Required | None
```
Use MuleSoft for the company, but Mule for the runtime engine.

Author: It is apparently not MuleSoft, but Anypoint Service Mesh license. I changed it. Thanks!

> | `license` | The Mule license you obtained as part of the prerequisites. Provide the path to the location where you license is stored. | Required | N/A

Reviewer 2: I think we need to put the Service Mesh licence (not mule)

Author: Done! Thanks!

| `platformUri` | The URL of the proxy.| Optional | https://anypoint.mulesoft.com
| `proxy.host` | The name of the proxy host. | Optional | default

> | `proxy.protocol`| The protocol used by the proxy, for example, HTTP or WSDL. | Optional | default

Reviewer 3: proxy.protocol options are only "http" or "https"

Reviewer 2: Remove `default`

Author: So what is the default value?

> | `proxy.protocol`| The protocol used by the proxy, for example, HTTP or WSDL. | Optional | default

Reviewer: It does not make sense that the value for the Dafault Value column is "default" in these two table rows.

Author: Yes, confirmed! Thanks!

> | `proxy.port`| The port number used by the proxy. | Optional | default

Reviewer 3: Properties were renamed to maintain camel case standard

proxyProtocol
proxyHost
proxyPort
proxyUser
proxyPassword

Author: Done! Thanks!

| `proxy.user` | The username for the proxy authentication. | Optional | default

> | `proxy.password`| The password for the proxy authentication. | Optional | default

Reviewer: ditto
Please fix/clarify

Author: Done! Thanks!

|===
+
Anypoint Service Mesh installs within few minutes and displays a status: 
+
image::service-mesh-installer-output.png[Example Installation Output]
+

> If an installation error occurs, the installer times out and uninstalls all previously installed Service Mesh components.

Reviewer: ```suggestion
If an installation error occurs, the installer times out and uninstalls all previously installed Anypoint Service Mesh components.
```

Author: Done! Thanks!

. Verify the status of all pods:
+
`kubectl get pods -n service-mesh`
+

> Anypoint Service Mesh is installed and a status of all pods is displayed:

Reviewer: ```suggestion
Anypoint Service Mesh is installed and the status of all pods is displayed:
```

Author: Done! Thanks!
